Remove outdated information from Regex docs (#17550)
dotnet/docs: docs/standard/base-types/compilation-and-reuse-in-regular-expressions.md

## Compiled Regular Expressions  
 By default, the regular expression engine compiles a regular expression to a sequence of internal instructions (these are high-level codes that are different from Microsoft intermediate language, or MSIL). When the engine executes a regular expression, it interprets the internal codes.  
  
 If a <xref:System.Text.RegularExpressions.Regex> object is constructed with the <xref:System.Text.RegularExpressions.RegexOptions.Compiled?displayProperty=nameWithType> option, it compiles the regular expression to explicit MSIL code instead of high-level regular expression internal instructions. This allows .NET's just-in-time (JIT) compiler to convert the expression to native machine code for higher performance.  The cost of constructing the <xref:System.Text.RegularExpressions.Regex> object may be higher, but the cost of performing matches with it is likely to be much smaller.

>  An alternative is to use precompiled regular expressions. You can compile all of your expressions into a reusable DLL by using the <xref:System.Text.RegularExpressions.Regex.CompileToAssembly%2A> method. This avoids the need to compile at runtime while still benefiting from the speed of compiled regular expressions.  

Reviewer: The next chapter is talking about precompiled regular expressions as well, but with a different meaning.

Change "Your application can take advantage of precompiled regular expressions in ..." to "Your application can reuse  regular expressions in ..."  ?

  
## The Regular Expressions Cache  
 To improve performance, the regular expression engine maintains an application-wide cache of compiled regular expressions. The cache stores regular expression patterns that are used only in static method calls. (Regular expression patterns supplied to instance methods are not cached.) This avoids the need to reparse an expression into high-level byte code each time it is used.  
  
 The maximum number of cached regular expressions is determined by the value of the `static` (`Shared` in Visual Basic) <xref:System.Text.RegularExpressions.Regex.CacheSize%2A?displayProperty=nameWithType> property. By default, the regular expression engine caches up to 15 compiled regular expressions. If the number of compiled regular expressions exceeds the cache size, the least recently used regular expression is discarded and the new regular expression is cached.  
  
 Your application can reuse regular expressions in one of the following two ways:  
  

> - By using a static method of the <xref:System.Text.RegularExpressions.Regex> object to define the regular expression. If you are using a regular expression pattern that has already been defined in another static method call, the regular expression engine will retrieve it from the cache, if it's still available. If not, or if it's not still available, the engine will compile the regular expression and add it to the cache.  

Reviewer 2: I think this makes it clearer.

```suggestion
- By using a static method of the <xref:System.Text.RegularExpressions.Regex> object to define the regular expression and add it to the cache. If you're using a regular expression pattern that has already been defined by another static method call, the regular expression engine will try to retrieve it from the cache. If it's not available in the cache the engine will compile the regular expression and add it to the cache.
```

Author: Thanks, but I personally think that makes it less clear.  The first sentence states that using the static method is going to add it to the cache, period. But it only adds it to the cache if it's not already there.

Reviewer 2: OK. I updated the text to remove that intro clause. How is this?

Author: Seems ok.
